feat: Arm model update: Inertia, scaling frames

toerholm opened the pull request:
@toerholm, Could you also add the missing description from the changelog here...

Commit:
Changed scaling directions for  radius bone, to align with new anatomical frame
Now radius and ulna is the using same scaling node

Commit:
added file used for doing rigidbody transformations between shoulder and trunk
this allows having shared scaling node for trunk and shoulder girdle (clavicula+ scapula)

Commit:
added generic bodymass correction class

Commit:
File used for inertia and mass distribution

Commit:
New def of anatomical frames
New def of scaling nodes for clavicula and scapula
Slightly altered pronation/supination axis
Introduction of inertia calculation based on a mesh class
...

Commit:
code clean up

Commit:
calculation of conoid lig based on equal scaling of scapula and clavicula

Commit:
nodes used for lower arm joints has been renamed and some has been slightly altered... joints config is in principle not changed

Commit:
removed the old joint config from code

Commit:
added switch for having a simple universal joint in the wrist  and added inertia scaling in the hand

Commit:
code clean up

Commit:
new initial position calculation based on renamed joint nodes..

Commit:
added inclusion of arm inertia calculation

Commit:
added code for universal wrist joint
modified code used for lower arm joints, nodes was renamed
added code for postural measures on clavicula and scapula

Commit:
code clean up

Commit:
CODE clean up

Commit:
code added for universal wrist joint option

Commit:
added data for points used to position clavicula and scapula
adjusted em_pos and el_pos
added data points for hand seg

Commit:
code clean up

Commit:
fixed holes in bones for inertia calculation

Commit:
added points for scapula and clavicula position

Commit:
added a switch around mass check in case of scaling none is used.. then the mass does not get adjusted

Commit:
added classes for creating mesh objects
Two sphere cone Mesh
Axis symmetric Mesh
Sphere Mesh

Commit:
added safeguard around mesh class

Commit:
fix endif bug

Commit:
added including the meshclass

Commit:
added switch for driver selection
This allows overriding FreePostureMove and FreePostureStatic by something user defined.

Commit:
removed markerprotocol file... it is taken from a shared location and this local file is not needed

Commit:
removed marker protocol files not being used anymore...

Commit:
changed scaling on radius to replicate ulna scaling they share scaling function now

Commit:
fixed symmetry issue in mannequin vals

Commit:
fixed symmetry issue in armrest

Commit:
removed reaction on a rhythm btw thigh and spine causing assym... and wrong results...

Commit:
fixed arm markers to match arm changes

Commit:
changed marker protocol to work with new arm

Commit:
changed humerus isb system ...

Commit:
changed the thickness of cyl... to avoid it to create a "bump" by sticking out on ellipsoid surf for deltoid... should possibly also be shorter..

Commit:
added harcoded vals for number of points to use... needs to be like this unfortunately because a scaling may change number of points already allocated... and cause an issue..

Commit:
changed naming

Commit:
added symmetry test for these symmetric application models

Commit:
code which can be included into applicatio models to test for symmetry

Commit:
added code for mesh to use a fixed number of points instead of trying to distribute points automatically... automatic distribution will break when scaling a model may lead to different allocation of points...

melund committed:
Merge commit '23b4b8c64143e6fd780a25982cf0c47d8657727a' into Inertia_wrok

melund committed:
Merge commit '384f146158f41bb25c379430a6f8e55807b99775' into Inertia_wrok

melund committed:
Merge branch 'ammr4-beta' into Inertia_wrok

Commit:
fixed name for include file

Commit:
added test model for arm inertia and scaling

melund commented on Body/AAUHuman/Arm/Interface.any:
```diff
@@ -37,16 +37,42 @@ AnyKinMeasureOrg SternoClavicularAxialRotation ={
+//This measure will be zero when SC protraction and SternoClavicularProtraction is -23 and SternoClavicularElevationVec is 11.5
+AnyKinMeasureOrg SternoClavicularProtraction_Postural = {
+  AnyKinRotational SCRot = {
+    AnyRefNode &thorax = .....Trunk.Segments.ThoraxSeg.ScalingNode;
```
@toerholm. Using `ThoraxSeg` here will not work when the detailed thorax is enabled.

melund commented on Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any:
```diff
@@ -157,19 +171,24 @@ AnyFolder SymmetryCheck= {
         };
       };
-      #if BM_LEG_MUSCLES_BOTH == OFF
+      #if  ((BM_LEG_MUSCLES_RIGHT != OFF) & (BM_LEG_MUSCLES_LEFT != OFF))
```
Why do you check for the leg muscles here? Should it not be the arm muscles. I guess it should have been: 
```suggestion
      #if  ((BM_ARM_MUSCLES_RIGHT == OFF) & (BM_ARM_MUSCLES_LEFT == OFF))
```
I think there is bug, where AnyBody crashes, when the class template `ComparePtrArrays` gets finds an empty array. But fixing the BM check should prevent it from including that code unless there are joint-muscles in the model.

toerholm committed:
Update Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any

Co-authored-by: Morten Enemark Lund <melund@gmail.com>
Signed-off-by: toerholm <st@anybodytech.com>

Commit:
removed debug visuals

Commit:
replaced scalingnode on thoraxseg with pelvisseg this makes flexible spine model work again

Commit:
replaced scalingnode on thoraxseg with pelvisseg

Commit:
Merge branch 'Inertia_wrok' of https://github.com/AnyBody/AMMR4-Beta into Inertia_wrok

melund commented on Body/AAUHuman/Arm/Seg.any:
```diff
@@ -1,72 +1,96 @@
-    AnyVec3 StandardsRel =.Data.NeutralPositionOnPelvis_pos;
-    //sRel=.Scale(StandardsRel*.Mirror);
+  r0 = ....Trunk.Segments.SternumSeg.r0 + ....Trunk.Segments.SternumSeg._SIDE_.scj.sRel*....Trunk.Segments.SternumSeg.Axes0'  - scj.sRel*Axes0';
+  Axes0= ....Trunk.Segments.ThoraxSeg.Axes0*Scale.T0.RigidBodyTransform.rot';
```
```suggestion
  Axes0= ....Trunk.Segments.PelvisSeg.Axes0*Scale.T0.RigidBodyTransform.rot';

```

melund commented on Body/AAUHuman/Arm/Seg.any:
```diff
@@ -83,67 +107,88 @@ AnySeg Clavicula ={
-  AnyFunTransform3D &Scale=..Scaling.Scapula.ScaleFunction;
-  Scale = { AnyFunTransform3D& GeomScale = .Scale; };
+  r0 = .Clavicula.acj.sRel*.Clavicula.Axes0'+.Clavicula.r0 -  acj.sRel*Axes0';
+  Axes0= ....Trunk.Segments.ThoraxSeg.Axes0*Scale.T0.RigidBodyTransform.rot';
```
```suggestion
  Axes0= ....Trunk.Segments.PelvisSeg.Axes0*Scale.T0.RigidBodyTransform.rot';

```

melund commented:
@toerholm. You need to look at the wrist flexion measure in the interface folder. I think it measures the wrong thing now. So, models where we drive the wrist flexion are 'crooked'

Commit:
changed thoraxseg to pelvisseg

melund commented on Tests/Body/ShoulderArm/Test_LoadPos_Scaling_Inertia/Test_LoadPos_Scaling_Inertia.Main.any:
```diff
@@ -0,0 +1,500 @@
+#include "../libdef.any"
+//#define WRIST_EXPERIMENTAL
+#include "Class_MorphingPipeline.any"
+#include "<AMMR_TOOLS>/ModelUtilities\SolidModeling/MeshClasses.any"
+
+
+
+//fatal_warnings = [""]
+//define = (
+//[
+//    {'SCALING_TYPE': 'NONE'},
+//    {'SCALING_TYPE': 'LINEAR'},
+//    {'SCALING_TYPE': 'RBF_SCALING'}
+//], 
+//[
+//    {'RIGHT_LEFT_SCALE_SYM': 'On'},
+//    {'RIGHT_LEFT_SCALE_SYM': 'Off'}
+//] 
+//)
+
```
The python header needs to be the very first thing in the file: 

```suggestion
//fatal_warnings = [""]
//define = (
//[
//    {'SCALING_TYPE': 'NONE'},
//    {'SCALING_TYPE': 'LINEAR'},
//    {'SCALING_TYPE': 'RBF_SCALING'}
//], 
//[
//    {'RIGHT_LEFT_SCALE_SYM': 'On'},
//    {'RIGHT_LEFT_SCALE_SYM': 'Off'}
//] 
//)

#include "../libdef.any"
//#define WRIST_EXPERIMENTAL
#include "Class_MorphingPipeline.any"
#include "<AMMR_TOOLS>/ModelUtilities\SolidModeling/MeshClasses.any"






```

melund commented on Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any:
```diff
@@ -144,7 +155,10 @@ AnyFolder SymmetryCheck= {
+          #if BM_LEG_MODEL != _LEG_MODEL_LEG_
+            AnyKinCoM Talus = {AnySeg &ref= Main.HumanModel.BodyModel.Right.Leg.Seg.Talus;};
+          #endif
           AnyKinCoM Foot = {AnySeg &ref= Main.HumanModel.BodyModel.Right.Leg.Seg.Foot;};
```
@toerholm: The foot is a AnyFolder when the detailed or 3-segment foot is used. So this will not work. It is the same issue on line 177.

@divyaksh-chander. What is best way around this? What kind of code/syntax will work in all cases?

divyaksh-chander replied:
We should use Foot.Talus. That will work for all the cases.

melund replied:
Here the code is refering to both `Seg.Talus` and `Seg.Foot` just below. What should the `Seg.Foot` reference be?

melund commented on Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any:
```diff
@@ -235,7 +259,7 @@ AnyFolder SymmetryCheck= {
           Array1 = ObjSearchRecursive("Main.HumanModel.BodyModel.Trunk.Abdominal", "*.Right.*.r");
           Array2 = ObjSearchRecursive("Main.HumanModel.BodyModel.Trunk.Abdominal", "*.Left.*.r");
-          MirrorMatrix = diag({1,1,-1});
+         MirrorMatrix = ...MirrorMatrix;
```
I needed to add 4 dots to make it load in this model: 
`Tests\Symmetry\test_TrunkDynamicSymmetryCheck.Main.any`
If this is not always needed, then there are some "include" problems here: 
```suggestion
           MirrorMatrix = ....MirrorMatrix;

```

melund commented on Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any:
```diff
@@ -339,8 +363,8 @@ AnyFolder SymmetryCheck= {
           Array2 = ObjSearchRecursive("Main.HumanModel.BodyModel.Trunk.Joints.RibCage.Left", "*.Fout");
-          MirrorMatrix = diag({1, 1, -1, 1, 1, -1});
-        };
+         MirrorMatrix = diag(arrcat(..MirrorPlane,..MirrorPlane));
```
I needed extra dots here as well: 

```suggestion
          MirrorMatrix = diag(arrcat(...MirrorPlane, ...MirrorPlane));

```

melund commented:
We need to update the hand and forearm marker-positions in all the MoCap models. Otherwise, they will fail with weird kinematic errors.

melund commented:
Besides what I already mention here is a list of failing trials: 

```
______ AnyBody Simulation: test_FreePostureFullBodyShoulderRhythmMove_0 _______
[gw3] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Applications\test_FreePostureFullBodyShoulderRhythmMove.any
Special model configuration:
  #define TEST_NAME "test_FreePostureFullBodyShoulderRhythmMove_0"
Errors:
> ERROR(SCR.PRS17) :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\MuscleSymmetryCheck.any(46)  :
  'ref'  :  Reference of class 'AnySeg' cannot refer to an object of class 'AnyFolder'
> ERROR(SCR.PRS17) :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\MuscleSymmetryCheck.any(60)  :
  'ref'  :  Reference of class 'AnySeg' cannot refer to an object of class 'AnyFolder'
> ERROR(SCR.EXP0) :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\CheckFullBody.any(30)  :
  'ArmPositions'  :  Error in evaluation. Please refer to the following error messages for details ...
> ERROR(SCR.EXP.FUN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\CheckFullBody.any(30)  :
  Defined at :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\CheckFullBody.any(3)  :   'ArmPositions' :
  'assert' failed : Arm model positions asymmetric


____________________ AnyBody Simulation: test_Abdominal_0 _____________________
[gw1] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Body\Trunk\Abdominal\test_Abdominal.any
Special model configuration:
  #define TEST_NAME "test_Abdominal_0"
Errors:
> ERROR(OBJ.MCH.MUS4) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Trunk\EvaluateJointStrengthStudy.any(161)  :
  'Study.InverseDynamics'  :  Muscle recruitment solver :  maximum number of setbacks occured


________________ AnyBody Simulation: test_BikeModel-FullBody_0 ________________
[gw3] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Applications\test_BikeModel-FullBody.any
Special model configuration:
  #define TEST_NAME "test_BikeModel-FullBody_0"
Errors:
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Application\Examples\BikeModel\BikeModel-FullBody.main.any(81)  :
  'Study.InitialConditions'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.MUS8) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\Muscle-parameters-shoulder.any(1832)  :
  'serratus_anterior_4'  :  Muscle model for muscle :  'serratus_anterior_4' :  Bad muscle activity calculation: Muscle
  force and strength leads to NaN.
> ERROR(OBJ.MCH.MUS8) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\LegTLEM\MusMdl3E_2.any(1902)  :   'IliacusMid2Par'  :
  Muscle model for muscle :  'IliacusMid2' :  Bad muscle activity calculation: Muscle force and strength leads to NaN.
> ERROR(OBJ.MCH.MUS8) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\Muscle-parameters-shoulder.any(1832)  :
  'serratus_anterior_4'  :  Muscle model for muscle :  'serratus_anterior_4' :  Bad muscle activity calculation: Muscle
  force and strength leads to NaN.
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Application\Examples\BikeModel\BikeModel-FullBody.main.any(81)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints


___________ AnyBody Simulation: test_StandingModelScalingDisplay_4 ____________
[gw3] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Applications\test_StandingModelScalingDisplay.any
Special model configuration:
  #define BM_SCALING 9
  #define TEST_NAME "test_StandingModelScalingDisplay_4"
Errors:
> ERROR(SCR.EXP0) :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\CheckFullBody.any(30)  :
  'ArmPositions'  :  Error in evaluation. Please refer to the following error messages for details ...
> ERROR(SCR.EXP.FUN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\CheckFullBody.any(30)  :
  Defined at :   D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Symmetry\DynamicTest\Model\CheckFullBody.any(3)  :   'ArmPositions' :
  'assert' failed : Arm model positions asymmetric


______________ AnyBody Simulation: test_EvaluateJointStrength_0 _______________
[gw2] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Applications\test_EvaluateJointStrength.any
Special model configuration:
  #define BM_LEG_MODEL 0
  #define BM_ARM_LEFT 1
  #define BM_ARM_RIGHT 1
  #define BM_TRUNK_MUSCLES 0
  #define TEST_NAME "test_EvaluateJointStrength_0"
Errors:
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 7 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.InitialConditions'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.InitialConditions'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 7 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.InitialConditions'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.InitialConditions'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Arm\EvaluateJointStrengthStudy.any(126)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints


____________________ AnyBody Simulation: test_WheelTurn_0 _____________________
[gw1] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Applications\test_WheelTurn.any
Special model configuration:
  #define TEST_NAME "test_WheelTurn_0"
Errors:
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-Beta\Application\Examples\WheelTurn\WheelTurn.Main.any(46)  :
  'Study.Kinematics'  :  Kinematic analysis failed in time step 88 : Failed to solve position constraints


______________ AnyBody Simulation: test_EvaluateJointStrength_3 _______________
[gw1] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Applications\test_EvaluateJointStrength.any
Special model configuration:
  #define BM_LEG_MODEL 0
  #define BM_ARM_LEFT 1
  #define BM_ARM_RIGHT 1
  #define BM_TRUNK_MUSCLES 1
  #define TEST_NAME "test_EvaluateJointStrength_3"
Errors:
> ERROR(OBJ.MCH.MUS4) :   D:\a\AMMR4-Beta\AMMR4-Beta\Body\AAUHuman\Trunk\EvaluateJointStrengthStudy.any(161)  :
  'Study.InverseDynamics'  :  Muscle recruitment solver :  maximum number of setbacks occured


_________ AnyBody Simulation: test_SpinePressureStandingLiftFlexed_0 __________
[gw0] win32 -- Python 3.12.10 D:\a\AMMR4-Beta\AMMR4-Beta\.pixi\envs\test\python.exe
Main file:
  D:\a\AMMR4-Beta\AMMR4-Beta\Tests\Applications\test_SpinePressureStandingLiftFlexed.any
Special model configuration:
  #define TEST_NAME "test_SpinePressureStandingLiftFlexed_0"
Errors:
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-
  Beta\Application\Validation\WilkeSpinalDiscPressure\SpinePressureStandingLiftFlexed\SpinePressureStandingLiftFlexed.ma
  in.any(73)  :   'Study.InitialConditions'  :  Kinematic analysis failed in time step 0 : Failed to solve position
  constraints
> ERROR(OBJ.MCH.KIN3) :   D:\a\AMMR4-Beta\AMMR4-
  Beta\Application\Validation\WilkeSpinalDiscPressure\SpinePressureStandingLiftFlexed\SpinePressureStandingLiftFlexed.ma
  in.any(73)  :   'Study.Kinematics'  :  Kinematic analysis failed in time step 0 : Failed to solve position constraints

````

Commit:
changed pelvisseg to sternum seg... this was a mistake in a recent commit...

Commit:
fixed symmetry issue on left radius scaling

Commit:
fixed wrong type

Commit:
fixed references to foot segs to work in more combinations

melund committed:
Update Tests/Body/ShoulderArm/Test_LoadPos_Scaling_Inertia/Test_LoadPos_Scaling_Inertia.Main.any

Signed-off-by: Morten Enemark Lund <melund@gmail.com>

melund commented on Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any:
```diff
@@ -273,7 +297,7 @@ AnyFolder SymmetryCheck= {
             Array1 = .right;
             Array2 = .left;
-            MirrorMatrix = diag({1,1,-1});
+          MirrorMatrix = ...MirrorMatrix;
```
```suggestion
            MirrorMatrix = ....MirrorMatrix;

```

melund committed:
Update Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any

Signed-off-by: Morten Enemark Lund <melund@gmail.com>

melund committed:
Update Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any

Signed-off-by: Morten Enemark Lund <melund@gmail.com>

melund committed:
Update Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any

Signed-off-by: Morten Enemark Lund <melund@gmail.com>

melund committed:
fix reference to MirrorMatrix

melund commented on Tests/Symmetry/DynamicTest/Model/MuscleSymmetryCheck.any:
```diff
@@ -261,7 +286,7 @@ AnyFolder SymmetryCheck= {
         ComparePtrArrays rib_inertia_pos() = {
```
@toerholm, This code fails as rib Inertia objects doesn't have a `r` value.

melund replied:
I fixed it.

melund committed:
fix: Compare sCoM symmetry for rib inertia pos as they have no `r` value

melund committed:
Fix arm rest symmetry problems

melund committed:
fix: quote arguments in header

Commit:
corrected deviation axis

Commit:
moved the code for axis calculation to the wrist seg... this is easier when having to switch btw hand models

Commit:
Merge branch 'Inertia_wrok' of https://github.com/AnyBody/AMMR4-Beta into Inertia_wrok

melund committed:
More quoting

melund committed:
Merge branch 'Inertia_wrok' of https://github.com/AnyBody/AMMR4-Beta into Inertia_wrok

melund committed:
fix: Ignore unimportant errors in test

melund committed:
Only fail on warnings

Commit:
fixed a symmetry issue in wristjoint seg

Commit:
Merge branch 'Inertia_wrok' of https://github.com/AnyBody/AMMR4-Beta into Inertia_wrok

melund committed:
Fix reference names in "BodyModelWihoutMuscles"

melund committed:
Fix handmarker protocol and compatability between simple and detailed hand.

Commit:
fixed left hand symmetry issue

Commit:
Merge branch 'Inertia_wrok' of https://github.com/AnyBody/AMMR4-Beta into Inertia_wrok

Commit:
Reworked the shoulder rhythm
It now ignores negative elevation planes there was no data for these

A starting guess on the ThoraxHumerus rotational measure that has a positive value for elevation has been added, this makes is more likely that among the rotational solutions it will pick the solution with a positive elevation.

Additionally it the starting guess is not enough to ensure positive elevation the ShoulderRhythm  is using a "processed" version of the thoraxHumerus measure.
This processed version flips all signs if elevation plane is negative

The rhythm code is also restructured to be wrapped into AnyKinEq

melund committed:
Merge remote-tracking branch 'origin/ammr4-beta' into Inertia_wrok

melund committed:
feat: Add conditional compilation for GHRot positive elevation initialization

melund committed:
fix: Correct syntax for excluding shoulder rhythm

melund commented:
@toerholm. We also need a section in the release notes about the all the changes introduced in this PR. 

It should include the following: 

* Short explanation of the new arm structure
* The new 'structure' of the shoulder rhythm - The fact that it is now a single driver called "shoulder rhythm"
* The updates the inertia of the arms

melund committed:
Merge branch 'ammr4-beta' into Inertia_wrok

melund commented:
@toerholm. I have updated the marker protocol for the ADL gait model, to match all the changes to the arm coordinate systems. 

Do you want me to add directly to this PR?

melund committed:
Merge remote-tracking branch 'origin/ammr4-beta' into Inertia_wrok

Commit:
resolve merge issues

melund committed:
Merge remote-tracking branch 'origin/ammr4-beta' into Inertia_wrok

melund committed:
feat: Update marker protocol and model files from rehazenter model

melund committed:
feat(adlgait): Update generated trial files

melund committed:
Update anyset file for first ADL gait beta example

melund committed:
Ignore errors due to markers not in dynamic trials

melund committed:
feat: Expand ignore_errors list in Test_ADL_Gait.any

melund committed:
Merge remote-tracking branch 'origin/ammr4-beta' into Inertia_wrok

melund committed:
Merge branch 'ammr4-beta' into Inertia_wrok

melund committed:
Split arm symmetry test between hand and arm

melund committed:
Ignore hand symmetry test

melund commented:
@toerholm 

The `RadioCarpalDeviation.ARel` node in the detailed hand is not correct. I have tried to fix it, but with no luck.

I have ignored the error for now, if you want to fix the problem in the hand model branch instead. 

- [ ] I still think we need a changelog description, and entry in the migration guide about these changes. 

[`Body\AAUHuman\Hand\HandSeg.any` line 380](https://github.com/AnyBody/AMMR4-Beta/blob/a3d4372710e110bf37bff7a00452e2fed9e04c56/Body/AAUHuman/Hand/HandSeg.any#L380)

```
  AnyRefNode RadioCarpalDeviation = {
    // wrist joint is defined from the two epicondyle bony landmarks
    sRel =.Scale(.Data.RadioCarpalDeviation_pos*.Mirror +{0.015,0.0,0}*.Mirror);
    //Note first rotations is because of difference in simple and detailed hans seg coordinate systems
    ARel =RotMat(pi*0.5,z)* RotMat(-pi*0.5,x)* ...WristJointSeg.RadioCarpalFlexion.ARel*RotMat(0.5*pi,z)*RotMat(pi,x)*RotMat((1-....Sign)*pi*0.5,x);  //note here we re-use the axis from RadioCarpalFlexion but flip it around this ensures the flexion and deviation axes are orthogonal
  };
```

melund committed:
fix test triggers

melund committed:
Merge branch 'ammr4-beta' into Inertia_wrok

Commit:
removed unused values and scj_y point for clavicula

Commit:
added changelog to documentation

Commit:
clean up

Commit:
class used to draw a colored coordinate system

Commit:
Model parameters cleaned up removing corrections and reduce digits

Commit:
moved coordinate system for SC joint to have a better position .. the orientation is the same

Commit:
changed comments

Commit:
added code allowing visualization of anatomical frames to be seen
changed code for sc joitn related to flip os scj_y to be downwards

Commit:
syntax

Commit:
added description and images of anatomical frames

Commit:
syntax

Commit:
added a guard on DriverSelection

Commit:
Merge branch 'Inertia_wrok' of https://github.com/AnyBody/AMMR4-Beta into Inertia_wrok

Commit:
removed visualizations of anatomical frames, they trigger sym test

melund committed:
Syncronize ADL gait model, to minimize git diffs

melund committed:
Minor formatting fixes
